apps/glusterfs/brick_allocate_test.go

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"testing"
+	"os"
+
+	"github.com/boltdb/bolt"
+	"github.com/heketi/tests"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+)
+
+func TestClusterDeviceSource(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	// Create the app
+	app := NewTestApp(tmpfile)
+	defer app.Close()
+
+	err := setupSampleDbWithTopology(app,
+		2,    // clusters
+		4,    // nodes_per_cluster
+		4,    // devices_per_node,
+		1*TB, // disksize)
+	)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	interfaceCheck := func(dsrc DeviceSource) DeviceSource {
```

Comment:
When there was no other code using the ClusterDeviceSource it was a simplistic way to check that it conformed to the DeviceSource interface. If I had made a mistake and they didn't match the test would fail to compile. Its not as important now that other code actually exists. Do you think I should drop it or add a comment?

Reply:
Let us have a comment for now. I would like to merge this patch today. Leaving the code here would allow @obnoxxx to give his comment on this later if he thinks otherwise.